server/src/test/java/org/opensearch/discovery/HandshakingTransportAddressConnectorTests.java

```diff
@@ -174,26 +173,21 @@ public void testLogsFullConnectionFailureAfterSuccessfulHandshake() throws Excep
 
         FailureListener failureListener = new FailureListener();
 
-        MockLogAppender mockAppender = new MockLogAppender();
-        mockAppender.start();
-        mockAppender.addExpectation(
-            new MockLogAppender.SeenEventExpectation(
-                "message",
-                HandshakingTransportAddressConnector.class.getCanonicalName(),
-                Level.WARN,
-                "*completed handshake with [*] but followup connection failed*"
-            )
-        );
         Logger targetLogger = LogManager.getLogger(HandshakingTransportAddressConnector.class);
-        Loggers.addAppender(targetLogger, mockAppender);
+        try (MockLogAppender mockAppender = MockLogAppender.createForLoggers(targetLogger)) {
+            mockAppender.start();
```

Comment:
This left over appender.start() should be removed.

Reply:
Good catch!